plugins/autosuspend.koplugin/main.lua

```diff
@@ -64,10 +64,14 @@ function AutoSuspend:_schedule(shutdown_only)
     -- On devices with an auxiliary battery, we only care about the auxiliary battery being charged...
     local powerd = Device:getPowerDevice()
     if Device:hasAuxBattery() and powerd:isAuxBatteryConnected() then
-        is_charging = powerd:isAuxCharging()
+        is_charging = powerd:isAuxCharging() and not powerd:isAuxCharged()
     else
-        is_charging = powerd:isCharging()
+        is_charging = powerd:isCharging() and not powerd:isCharged()
     end
+    -- We *do* want to make sure we attempt to go into suspend/shutdown again while *fully* charged, though.
+    -- FIXME: Suspend is tricky on Kobo, though.
```

Comment:
FWIW: In #8934 I use "onCharging" to detect a change in charging state. Maybe this helps.